.github/workflows/validate.yaml

```diff
@@ -0,0 +1,124 @@
+  schedule:
+    - cron:  '20 2 * * *'
+
+name: validate
+
+jobs:
+  db-tests:
+    runs-on: ${{ matrix.config.os }}${{ matrix.config.os-version }}
+
+    name: ${{ matrix.config.os }}${{ matrix.config.os-version }} (${{ matrix.config.r }}) ${{ matrix.config.desc }}
+
+    # Begin custom: services
+    # End custom: services
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - { os: ubuntu-, os-version: 18.04,   r: 'release', rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest" }
+
+    env:
+      R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
+      RSPM: ${{ matrix.config.rspm }}
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      # prevent rgl issues because no X11 display is available
+      RGL_USE_NULL: true
+      # Begin custom: env vars
+      # End custom: env vars
+
+    steps:
+      - name: Check rate limits
+        run: |
+          curl -s --header "authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/rate_limit
+        shell: bash
+
+      - uses: actions/checkout@v2
+
+      # Begin custom: before install
+      # End custom: before install
+
+      - uses: r-lib/actions/setup-r@v1
+        with:
+          r-version: ${{ matrix.config.r }}
+          http-user-agent: ${{ matrix.config.http-user-agent }}
+
+      - name: Install remotes
+        run: |
+          if (!requireNamespace("curl", quietly = TRUE)) install.packages("curl")
+          if (!requireNamespace("remotes", quietly = TRUE)) install.packages("remotes")
+        shell: Rscript {0}
+
+      - name: Query dependencies
+        if: runner.os != 'Windows'
+        run: |
+          saveRDS(remotes::dev_package_deps(dependencies = TRUE, type = .Platform$pkgType), ".github/depends.Rds", version = 2)
+          writeLines(sprintf("R-%i.%i", getRversion()$major, getRversion()$minor), ".github/R-version")
+        shell: Rscript {0}
+
+      - name: Restore R package cache
+        if: runner.os != 'Windows'
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.R_LIBS_USER }}
+          key: ${{ matrix.config.os }}${{ matrix.config.os-version }}-${{ hashFiles('.github/R-version') }}-2-${{ hashFiles('.github/depends.Rds') }}
+          restore-keys: ${{ matrix.config.os }}${{ matrix.config.os-version }}-${{ hashFiles('.github/R-version') }}-2-
+
+      - name: Install system dependencies
+        if: runner.os == 'Linux'
+        run: |
+          while read -r cmd
+          do
+            eval sudo $cmd
+          done < <(Rscript -e 'writeLines(remotes::system_requirements("ubuntu", "${{matrix.config.os-version}}"))')
+
+      # Begin custom: before install
+      # End custom: before install
+
+      - name: Install dependencies
+        run: |
+          remotes::install_deps(dependencies = TRUE, type = .Platform$pkgType)
+        shell: Rscript {0}
+
+      - name: Instrument R code for validation
+        run: |
+          sed -i -r '/INSTRUMENT: validate/ s/^( +)# /\1/g' R/*
```

Comment:
1. Are you adding this extra workflow so we can see how long the validation takes by comparison? 
1. Do we know in how many cases the validation is then executed in the following tests?
1. Are you using the general regex in the sed command because there might be more places where we want to use `validate_dm()` in the future?

Reply:
1. No
2. It's executed every time `new_dm3()` is called, except if it's called from `validate_dm()` -- to avoid infinite recursion
3. I don't worry about replacing multiple instances. However, we should check that the `sed` actually changes at least one file.